### PR TITLE
feat: erc20 wrapper

### PR DESCRIPTION
**Pull request by djm07073**
### Feat
- erc20 wrapper to support compatible ibc transfer from minievm to other vm( movevm/ wasmvm) chain 

## Release Notes

- **New Features**
  - Introduced the `ERC20Wrapper` contract for wrapping and unwrapping ERC20 tokens, enhancing interoperability with Cosmos via IBC.
  - Added support for querying the ERC20 wrapper address through a new method in the query server.
  - Expanded documentation for EVM hooks, including a new `Value` field in the message structure and examples for IBC transfers.
  - Added a new field to the genesis state for the ERC20 wrapper address.

- **Bug Fixes**
  - Improved error handling for ERC20 wrapper address retrieval.

- **Documentation**
  - Updated `README.md` for the `app/ibc-hooks` module to reflect changes in EVM contract execution and new field requirements.

- **Chores**
  - Enhanced clarity in comments and structure of initialization methods in the keeper package.

### CODE REVIEW

File: app/ibc-hooks/README.md [hunk 1]
```diff
@@ -71,6 +74,8 @@ msg := MsgCall{
  ContractAddr: packet.data.memo["evm"]["message"]["contract_addr"],
  // Hex encoded execution input bytes.
  Input: packet.data.memo["evm"]["message"]["input"],
+
+ Value: packet.data.memo["evm"]["message"]["value"]
```
Comment (beer-1) [inline app/ibc-hooks/README.md:78]: let's add newly added access list too

File: x/evm/keeper/genesis.go [hunk 2]
```diff
@@ -42,12 +43,23 @@ func (k Keeper) InitializeWithDecimals(ctx context.Context, decimals uint8) erro
...
 
-	// 2. deploy fee denom erc20 coins at genesis bootstrapping
+	// 2. Deploy fee denom ERC20 coins at genesis bootstrapping with decimals.
 	err = k.erc20Keeper.CreateERC20(ctx, params.FeeDenom, decimals)
 	if err != nil {
 		return err
 	}
 
+	// 3. Deploy and store the wrapper ERC20 factory contract for IBC transfers to the destination chain (not compatible due to 18 decimals).
+	_, wrapperAddr, _, err := k.EVMCreate2(ctx, types.StdAddress, code, nil, types.ERC20WrapperSalt)
```
Comment (beer-1) [inline x/evm/keeper/genesis.go:53]: need to load erc20 wrapper code

File: x/evm/contracts/erc20_wrapper/ERC20Wrapper.sol [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+    function _createERC20(
+        string memory name,
+        string memory symbol,
+        uint8 decimals
+    ) internal returns (address) {
+        ERC20 erc20 = new ERC20(name, symbol, decimals);
+
+        // register the ERC20 contract with the ERC20 registry
+        ERC20_REGISTRY_CONTRACT.register_erc20_from_factory(address(erc20));
```
[the remark this answers is not in the document]
Comment (djm07073) [reply x/evm/contracts/erc20_wrapper/ERC20Wrapper.sol:136]: i got it thank you

**Review by beer-1 — APPROVED** [review]
LGTM

**Pull request merged**